src/main/java/org/terasology/gf/generator/FloraProvider.java

```diff
@@ -61,13 +56,9 @@ public void process(GeneratingRegion region) {
 
         for (int x = minX; x <= maxX; x++) {
             for (int z = minZ; z <= maxZ; z++) {
-                int height = TeraMath.floorToInt(surface.getWorld(x, z));
-                // if the surface is in range, and if we are above sea level
-                if (facet.getWorldRegion().encompasses(x, height, z) && facet.getWorldRegion().encompasses(x, height + 1, z) && height >= seaLevel) {
-
-                    // if the block on the surface is dense enough
-                    if (density.getWorld(x, height, z) >= 0
-                            && density.getWorld(x, height + 1, z) < 0) {
```

Comment:
Why is this not required at all anymore?

Reply:
The SurfacesFacet should already contain this information, as it should only contain points where the density is such that the point is a surface.